cli/yamls.py

```diff
@@ -330,29 +57,126 @@ def to_yaml(cls, dumper, node):
-        # fix for the empty yml files
-        if confError.message == 'unconfigured':
-            return {}
-        else:
-            raise
+        raise exceptions.IRYAMLConstructorError(e, settings_file)
+
+
+def _get_most_nested_lookups(string_value):
+    """
+    Returns list of most nested lookups patterns from a given string (in
+    case of lookup in lookup)
+
+    :param string_value: String to search lookup patterns in it
+    :return: List containing lookup patterns
+    """
+    parser = re.compile('\{\{\s*\!lookup\s*[\w.]*\s*\}\}')
+    return parser.findall(string_value)
+
+
+def replace_lookup(lookups_dict):
+    """
+    Replaces all lookup pattern in a given dictionary (lookups_dict) with
+    the corresponding values
+
+    :param lookups_dict: Dictionary with data to replace
+    """
+    def yaml_walk(yaml_content, key_path=None):
+        an_iter = enumerate(yaml_content) if isinstance(yaml_content, list) \
+            else yaml_content.iteritems()
+
+        for idx_key, value in an_iter:
+            if hasattr(value, '__iter__'):
+                if key_path is None:
+                    next_key = str(idx_key)
+                else:
+                    next_key = key_path + '.' + str(idx_key)
+                yaml_walk(value, next_key)
+            elif isinstance(value, str):
+                yaml_content[idx_key] = _lookup_handler(
+                    value, lookups_dict, key_path)
+
+    yaml_walk(lookups_dict)
+
+
+def _lookup_handler(lookup_string, data_dict, key_path, visited=None):
+    """
+    Lookups handling mechanism
+
+    :param lookup_string: string containing the lookup pattern to be handled
+    :param data_dict: The complete data dictionary
+    :param key_path: Path to the value's key we currently handling
+    :param visited: List of already visited keys (infinite loop indicator)
+    :return:
+    """
+    if visited is None:
+        visited = []
+    visited.append(key_path)
+
+    while True:
+        lookups = _get_most_nested_lookups(lookup_string)
+        if not lookups:
+            break
+
+        for lookup in lookups:
+            lookup_key = re.search('(\w+\.?)+ *?\}\}', lookup)
+            lookup_key = lookup_key.group(0).strip()[:-2].strip()
+
+            if lookup_key in visited:
+                raise exceptions.IRInfiniteLookupException(", ".join(visited))
+
+            new_value = _lookup_handler(dict_get(data_dict, lookup_key),
+                                        data_dict, lookup_key, visited)
+
+            lookup_string = re.sub(lookup, new_value, lookup_string, count=1)
+
+    return lookup_string
+
+
+def dict_get(dic, key):
+    """
+    Gets the value of a nested key in a dictionary
+
+    :param dic: Dictionary containing the needed data
+    :param key: String representing full key path (Example: 'key.sub1.sub2')
+    :return: Value of a given key in the dictionary
+    """
+    def _dict_get(_dic, current_key, *keys):
```

Comment:
Please remove this inner method as there is no extra logic in having it as an inned method